utils/fs.go

```diff
@@ -33,11 +32,11 @@ func EnsureDirectory(path string, perm os.FileMode) error {
 	}
 	// file does not exist (or has been deleted)
 	if err == nil || os.IsNotExist(err) {
-		err = os.MkdirAll(path, perm)
+		err = os.Mkdir(path, perm)
```

Comment:
Shouldn't we still use `MkdirAll()` here? Otherwise we not expect the parent directory to exist